### PR TITLE
Add K6 Hipster Shop example

### PR DESCRIPTION
**Pull request by ztreinhart**
This PR adds the hipster shop manifests and an example experiment that uses [k6](https://k6.io) and the [k6 trial image](https://github.com/thestormforge/optimize-trials/tree/main/k6) to optimize the `frontend` component of the hipster shop. 

I also took this opportunity to rearrange the webserver/webapp examples, add READMEs at all levels of the folder structure, clean up some wording and typos, etc.

### Summary of changes:

- Renamed top level directory from `webserver` to `webapp` to reflect more specific/current terminology
- In the `webapp` directory, there is now a folder for each demo application: one for `voting-webapp`, one for `hipster-shop`
- In each demo application directory, the application description, manifests, etc, all now live under the `application/` directory, with manifests in `application/manifests`
- Parallel to the `application/` folder in the demo application directory, are folders for the specific experiments available, broken down by the performance testing tool they demonstrate: `locust` and `stormforge-perftest` under `voting-webapp` and `k6` under `hipster-shop`
- Added (hopefully useful) README.md files at all major levels of the folder structure.
- Cleaned up nomenclature: removed "-metrics" suffixes from some experiment names, standardized on "webapp" instead of "webserver" in READMEs, etc.
- Removed the `vhs-sf-perftest` experiment since `vhs` is no longer a maintained project and may not even still work.
- Fixed typos as I found them, minor rewording to improve clarity, etc.

That may not be everything, but it covers the major points.

### CODE REVIEW

File: webapp/hipster-shop/k6/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# k6 Example
+
+The StormForge Optimize Pro experiment in this folder uses [k6](https://k6.io) to generate load against the Online Boutique test application and track performance metrics through the course of an experimental trial. 
+
+The experiment utilizes a custom trial image based on [the K6 trial image](https://github.com/thestormforge/optimize-trials/tree/main/k6) distributed by StormForge.
```
Comment (tisba) [inline webapp/hipster-shop/k6/README.md:5]:
```suggestion
The experiment utilizes a custom trial image based on [the k6 trial image](https://github.com/thestormforge/optimize-trials/tree/main/k6) distributed by StormForge. 
```

File: webapp/hipster-shop/k6/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# k6 Example
+
+The StormForge Optimize Pro experiment in this folder uses [k6](https://k6.io) to generate load against the Online Boutique test application and track performance metrics through the course of an experimental trial. 
+
+The experiment utilizes a custom trial image based on [the K6 trial image](https://github.com/thestormforge/optimize-trials/tree/main/k6) distributed by StormForge. 
+This trial image is used as the basis of each trial job that Optimize Pro runs during an experiment.
+
+The custom trial image uses the K6 trial image as the base layer, and adds the test script and test data to the image for convenience.
```
Comment (tisba) [inline webapp/hipster-shop/k6/README.md:8]:
```suggestion
The custom trial image uses the k6 trial image as the base layer, and adds the test script and test data to the image for convenience. 
```

File: webapp/voting-webapp/application/README.md [hunk 3]
```diff
@@ -16,27 +16,27 @@ The application architecture is made of 5 components (see diagram below):
 
 ## Resources
 
-The resources in this directory aren't going to launch experiments in StormForge Optimize in and off themselves. For that, see [locust-metrics](../locust-metrics). Nevertheless, here is a short description for each of them.
+The resources in this directory aren't going to launch experiments in StormForge Optimize Pro in and of themselves. For that, see [locust-metrics](../locust-metrics). Nevertheless, here is a short description for each of them.
```
Comment (tisba) [inline webapp/voting-webapp/application/README.md:19]: probably just `lucust` now…

**Review by tisba — APPROVED** [review]

File: webapp/voting-webapp/application/README.md (hunk 3 above)
Comment (ztreinhart) [reply webapp/voting-webapp/application/README.md]: Fixed in https://github.com/thestormforge/examples/pull/77/commits/cd201e8f085af6da53edc1436f821be40a6f3997

**Review by tisba — APPROVED** [review]

**Pull request merged**